### PR TITLE
Add cutoff pressures to tian parameterization

### PR DESCRIPTION
**Pull request by danieldouglas92**
The polynomials used in the Tian parameterization for reactive two phase flow break down above certain pressures for each lithology, and the maximum bound water content becomes infinite. This PR ensures that the polynomial does not break down above these pressures by setting the pressure equal to the pressure just before the break down. An example of how this fix impacts the PT space for gabbro is shown below.

image

image

The python script that was used to determine these cutoff pressures can be found in the zip below. 
[Tian_water_content.py.zip](https://github.com/geodynamics/aspect/files/15491255/Tian_water_content.py.zip)

### CODE REVIEW

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (tjhei) [inline include/aspect/material_model/reactive_fluid_transport.h:184]: can you make this const?

File: source/material_model/reactive_fluid_transport.cc
Comment (tjhei) [inline source/material_model/reactive_fluid_transport.cc:72]: What is 1e-12? Can you introduce a variable for it and give it a name reflecting this?

File: source/material_model/reactive_fluid_transport.cc
Comment (tjhei) [inline source/material_model/reactive_fluid_transport.cc:73]:
```suggestion
          const double inverse_pressure = 1.0/pressure;
```

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (naliboff) [reply include/aspect/material_model/reactive_fluid_transport.h:184]: Are these cutoff pressures reported in the Tian et al. paper? If yes, can you add a sentence to the documentation above to indicate this?

**Review by naliboff — COMMENTED** [review]
@danieldouglas92 - Thanks for implementing this change. Reiterating my comment within the PR, are the cutoff values for each lithology reported in the Tian et al. paper, or did you determine them through testing? If the latter, would it be worth creating an input parameter where people can try adjusting the cutoff pressures?

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (danieldouglas92) [reply include/aspect/material_model/reactive_fluid_transport.h:184]: As far as I could see these values are not reported in the Tian et al., paper, but these values could definitely be made constant yes

File: source/material_model/reactive_fluid_transport.cc
Comment (danieldouglas92) [reply source/material_model/reactive_fluid_transport.cc:72]: It is just a small value that is close to 0 that prevents a division by 0 in the next line

Comment (naliboff) [reply source/material_model/reactive_fluid_transport.cc:73]: @tjhei - would it be more self-explanatory to use `numeric_limits::min()` instead of 1e-12, and/or add a comment?

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (naliboff) [inline include/aspect/material_model/reactive_fluid_transport.h:183]:
```suggestion
        // and 50 GPa for sediment. These cutoff pressures were determined by extending the pressure range in Tian et al. (2019) and observing where the maximum allowed water contents jump towards infinite values.
```

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (naliboff) [inline include/aspect/material_model/reactive_fluid_transport.h:184]: I would delete these two lines, but still leave the python script in the PR

**Review by naliboff — COMMENTED** [review]
@danieldouglas92 - Just a few additional comments.

File: include/aspect/material_model/reactive_fluid_transport.h
Comment (naliboff) [inline include/aspect/material_model/reactive_fluid_transport.h:185]: Remove this line?

**Review by naliboff — APPROVED** [review]
Good to go from my side once you address my last comment and @tjhei approves.

File: source/material_model/reactive_fluid_transport.cc
Comment (tjhei) [inline source/material_model/reactive_fluid_transport.cc:74]:
```suggestion
          // Introduce minimum pressure to avoid a division by 0.
          const double minimum_pressure = 1e-12;
          const double pressure = std::min(std::max(minimum_pressure, in.pressure[q]/1.e9), pressure_cutoffs[i]);
          const double inverse_pressure = 1.0/pressure;
```

**Review by tjhei — APPROVED** [review]

**Pull request merged**
